### PR TITLE
Fix authority fuzzing failures

### PR DESCRIPTION
**Pull request by dgquintas**
Introduced in #14615

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
  +0.0%     +48 [None]                                                  +216  +0.0%
  +7.9%     +48 src/core/ext/filters/http/client_authority_filter.cc     +48  +7.9%
       +20%     +43 (anonymous namespace)::init_channel_elem                 +43   +20%
       +14%      +5 [Unmapped]                                                +5   +14%
  +0.9%     +48 src/core/lib/surface/channel.cc                          +48  +0.9%
      +9.3%     +41 grpc_channel_create                                      +41  +9.3%
      +5.4%      +7 [Unmapped]                                                +7  +5.4%

  +0.0%    +144 TOTAL                                                   +312  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: src/core/ext/filters/http/client_authority_filter.cc [hunk 1]
```diff
@@ -97,8 +97,15 @@ grpc_error* init_channel_elem(grpc_channel_element* elem,
...
     abort();
   }
-  chand->default_authority = grpc_slice_from_copied_string(
-      grpc_channel_arg_get_string(default_authority_arg));
+  const char* default_authority_str =
+      grpc_channel_arg_get_string(default_authority_arg);
+  if (default_authority_str == nullptr) {
+    gpr_log(GPR_ERROR,
+            "GRPC_ARG_DEFAULT_AUTHORITY channel arg. must be a string.");
+    abort();
```
Comment (markdroth) [inline src/core/ext/filters/http/client_authority_filter.cc]: Instead of aborting here, how about returning a `grpc_error`?

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: src/core/ext/filters/http/client_authority_filter.cc (hunk 1 above)
Comment (dgquintas) [reply src/core/ext/filters/http/client_authority_filter.cc]: Good idea. Done.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                            FILE SIZE
 ++++++++++++++ GROWING                                              ++++++++++++++
   +18%    +112 src/core/ext/filters/http/client_authority_filter.cc    +112   +18%
       +54%    +118 (anonymous namespace)::init_channel_elem                +118   +54%
  +0.9%     +48 src/core/lib/surface/channel.cc                          +48  +0.9%
      +9.3%     +41 grpc_channel_create                                      +41  +9.3%
      +5.4%      +7 [Unmapped]                                                +7  +5.4%

 -+-+-+-+-+-+-+ MIXED                                                +-+-+-+-+-+-+-
  +0.0%     +88 [None]                                                   -88  -0.0%

  +0.0%    +248 TOTAL                                                    +72  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Pull request merged**